code/modules/research/xenobiology/crossbreeding/loyal.dm

```diff
@@ -0,0 +1,745 @@
+	attached.add_atom_colour(pick(colorlist), FIXED_COLOUR_PRIORITY)
+
+/obj/item/slimecross/loyal/red
+	colour = "red"
+	effect_desc = "Makes an item stronger when thrown, if it is covered in blood."
+	loyaleffect = /datum/component/loyal_effect/red
+
+/datum/component/loyal_effect/red
+	prefix = "blooded"
+	var/bloody = FALSE
+
+/datum/component/loyal_effect/red/Initialize(obj/item/attached_item)
+	. = ..()
+	START_PROCESSING(SSobj,src)
+
+/datum/component/loyal_effect/red/Destroy()
+	STOP_PROCESSING(SSobj,src)
+	return ..()
+
+/datum/component/loyal_effect/red/process()
+	if(attached.blood_DNA_length() && !bloody) //This would be much nicer if making something bloody was done through comsignals
+		change_blood()
+	else if(!attached.blood_DNA_length() && bloody)
+		change_blood()
+
+/datum/component/loyal_effect/red/proc/change_blood()
+	if(bloody)
+		bloody = FALSE
+		attached.throwforce -= 2
+	else
+		bloody = TRUE
+		attached.throwforce += 2
+
+/obj/item/slimecross/loyal/green
+	colour = "green"
+	effect_desc = "Makes an item able to be disguised as another item."
+	loyaleffect = /datum/component/loyal_effect/green
+
+/datum/component/loyal_effect/green
+	prefix = "disguised"
+
+/datum/component/loyal_effect/green/Initialize(obj/item/attached_item)
+	. = ..()
+	attached_item.verbs += /obj/item/proc/Disguise
+
+/obj/item/proc/Disguise(mob/user)
+	var/disguise_list //Possible disguises. If there is a common file of items by weight, let me know.
+	switch(w_class)
+		if(WEIGHT_CLASS_TINY)
+			disguise_list = list(/obj/item/pen, //things that are odd, but not immediately suspicious.
+								 /obj/item/lipstick, //Things that are Bond-like
+								 /obj/item/candle, //Because disguising vital things as mundane things
+								 /obj/item/soap //Is a very Bond thing to do
+								)
+		if(WEIGHT_CLASS_SMALL)
+			disguise_list = list(/obj/item/wrench,
+								 /obj/item/toy/cards/deck,
+								 /obj/item/clipboard,
+								 /obj/item/instrument/harmonica
+								)
+		if(WEIGHT_CLASS_NORMAL)
+			disguise_list = list(/obj/item/reagent_containers/food/snacks/grown/pineapple,
+								 /obj/item/mop,
+								 /obj/item/roller,
+								 /obj/item/extinguisher
+								)
+		if(WEIGHT_CLASS_BULKY)
+			disguise_list = list(/obj/item/storage/toolbox,
+								 /obj/item/shield/riot,
+								 /obj/item/storage/briefcase,
+								 /obj/item/gun/ballistic/shotgun/doublebarrel
+								)
+		if(WEIGHT_CLASS_HUGE)
+			disguise_list = list(/obj/item/katana, //Not quite mundane anymore.
+								 /obj/item/melee/baseball_bat,
+								 /obj/item/electropack,
+								 /obj/item/chair
+								)
+		if(WEIGHT_CLASS_GIGANTIC)
+			disguise_list = list(/obj/item/mecha_parts/part/honker_head, //There's barely anything of this size
+								 /obj/item/his_grace //Disguising as this is more of a laugh than a strategy
+								)
+	var/obj/item/disguise_type = pick(disguise_list)
+	var/obj/item/disguise = new disguise_type
```

Comment:
Any way to get the properties of disguise_type without making an instance of it?

Reply:
Also, Disguise should only be usable in hand. Same with compress in bluespace I guess. How do I check this?